vocabularies/Common.json

```diff
@@ -1289,6 +1289,95 @@
       "@Common.Experimental": true,
       "@Core.IsURL": true,
       "@Core.Description": "Base URL for WebSocket connections"
+    },
+    "GetRecommendations": [
+      {
+        "$Kind": "Function",
+        "$IsBound": true,
+        "@Common.Experimental": true,
+        "@Core.IsSignature": true,
+        "@Core.Description": "Computes AI-based recommendations for the bound entity and its related entities",
+        "@Core.LongDescription": "The function SHOULD compute the recommendations asynchronously, see [this diagram](../docs/recommendations.md).",
+        "$Parameter": [
+          { "$Name": "Entity", "$Type": "Edm.EntityType" },
+          {
+            "$Name": "Targets",
+            "$Type": "Common.RecommendationsTargetsType",
+            "@Core.Description": "Entities, properties or action/function parameters for which recommendations shall be retrieved",
+            "@Core.LongDescription": "Implementations of this function MAY omit this parameter.\n            If this parameter is omitted (by the implementation or in the invocation), all recommendations are retrieved.",
+            "@Core.OptionalParameter": {}
+          }
+        ],
+        "$ReturnType": {
+          "$Collection": true,
+          "$Type": "Common.PropertyRecommendationType",
+          "@Core.Description": "A collection of recommendations that differ in their `AIRecmddFldURL` property"
+        }
+      }
+    ],
+    "RecommendationsTargetsType": {
+      "$Kind": "ComplexType",
+      "@Common.Experimental": true,
+      "URLs": {
+        "$Collection": true,
+        "@Core.IsURL": true,
+        "@Core.Description": "URLs (relative to the request URL) that address entities, properties or action/function parameters for which recommendations shall be retrieved",
+        "@Core.LongDescription": "The request `SalesOrder('A')/ns.GetRecommendations?Targets={\"URLs\":[\"\",\"Items(10)\"]}`\n            retrieves recommendations for the entities `SalesOrder('A')` and `SalesOrder('A')/Items(10)`."
+      }
+    },
+    "PropertyRecommendationType": {
+      "$Kind": "ComplexType",
+      "@Common.Experimental": true,
+      "AIRecmddFldURL": {
+        "@Core.IsURL": true,
+        "@Core.Description": "URL (relative to the URL of the [`GetRecommendations`](#GetRecommendations) function request) addressing a property [OData-URL, section 4.6] that is targeted by this recommendation",
```

Comment:
Not sure how does the relative Url work in the following cases

- Editing an instance which has items, when the list is loaded there is one function call made to retrieve the recommendation & pass all the items in the UI as targets, for an example in a request like below

```js
SalesOrder('A')/ns.GetRecommendations?Targets={\"URLs\":[\"\",\"Items(10)\",\"Items(20)\",\"Items(30)\"]}
```

We will expect the results for SalesOrder A, Item(10), Item(20), Item(30) to return. So what is the expectation here

``` json
[
        {
            "target": "/CustomerPaymentTerms",
            "value": "",
            "text": "Cash only",
            "alternatives": [
                {
                    "value": "",
                    "score": 98
                },
                {
                    "value": "0009",
                    "score": 85
                }
            ]
        },
        {
            "target": "/_ShipToParty/BusinessPartner",
            "value": "69406910",
            "text": "Domestic NZ Supplier 1",
            "alternatives": [
                {
                    "value": "69406910",
                    "score": 99
                },
                {
                    "value": "45404510",
                    "score": 90
                },
                {
                    "value": "62100010",
                    "score": 85
                }
            ]
        },
        {
            "target": "/_Item(10)/Material",
            "value": "RMPI002",
            "text": "Input 1 AGENT 2",
            "alternatives": [
                {
                    "value": "RMPI002",
                    "score": 99
                },
                {
                    "value": "RMPI004",
                    "score": 90
                },
                {
                    "value": "RMPI003",
                    "score": 85
                }
            ]
        },
        {
            "target": "/_Item(20)/Material",
            "value": "RMPI002",
            "text": "Input 1 AGENT 2",
            "alternatives": [
                {
                    "value": "RMPI002",
                    "score": 99
                },
                {
                    "value": "RMPI004",
                    "score": 90
                },
                {
                    "value": "RMPI003",
                    "score": 85
                }
            ]
        }
    ]
```
_**Note: Here Item(30) doesn't have any recommendation. CustomerPaymentTerms is coming from Header node, BusinessPartner coming from 1..1 association _ShipToParty & Material comes from 1..n association _Item**_

Of course the names represent initial proposal and show updated names. @HeikoTheissen, Can you confirm?

Reply:
Costa's mail suggested a different format: `/SalesOrder('A')/_Item(20)/Material`.

I will now change the vocabulary to reflect the format in your example, although I don't understand the leading slash.

Reply:
Why don't we use the exact same format for `target` that we are using in warning and info messages for `target` and `additionalTargets`?

Reply:
And while we are at it, we could document this format here in addition to the internal wiki page where we currently hide it 😄

Reply:
@HeikoTheissen, 

- Currently we have coded based on the absolute path. But we could make this to relative URL for consistency & lesser repetition. So relative URL is Ok
- Leading slash is a mistake in my sample data.  So it would be "CustomerPaymentTerms" and not "/CustomerPaymentTerms". It applies in all other cases as well
- @ralfhandl, relative Url is something which we use even in Messages. It is then consistent